lib/compat/wordpress-6.0/register-webfonts-from-theme-json.php

```diff
@@ -58,7 +55,8 @@ function gutenberg_register_webfonts_from_theme_json() {
 		}
 	}
 	foreach ( $webfonts as $webfont ) {
-		wp_webfonts()->register_font( $webfont );
+		$id = isset( $webfont['id'] ) ? $webfont['id'] : gutenberg_generate_font_id( $webfont );
```

Comment:
The sole reason we're doing this ternary is that it might break themes already using the Webfont API and registering webfonts, but without specifying the font id.

The thing is, the Webfont API hasn't been made public yet, so we can do whatever we want with the signature... Including breaking changes such as this one.

### "Why are we making the `id` key required?"

The idea is that developers provide their own hashing mechanism so they can register IDs that works for them, since they're creating their own.

I do believe that adding the `gutenberg_generate_font_id` function is confusing. The reason why it's there is that no theme specifies the ID when registering a font through theme.json so we wanted to maintain it backward compatible. That's why this function is used there.

So, now that we're separating registration from enqueueing, I do think it makes sense for the developers to come with their own IDs, so that they can register and enqueue the fonts as they need instead of relying on an abstract mechanism that they're not aware of.